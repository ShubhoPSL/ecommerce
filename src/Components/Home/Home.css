.home-container {
  @apply flex justify-center items-center flex-col bg-yellow-300 h-[83.25vh] w-auto;
}

.home-body {
  @apply text-3xl 
    font-light;
}

.catchy-tagline {
  @apply text-4xl uppercase font-bold;
}

.shopping-btn {
  @apply inline-block px-3 py-3 bg-green-500 rounded my-4 text-justify font-bold;
}
