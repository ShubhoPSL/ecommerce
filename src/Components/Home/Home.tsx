import React from "react";
import { Link } from "react-router-dom";

function Home() {
  return (
    <div className="home-container">
      <p className="home-body">
        ShopKart is the fastest growing e-commerce platform of US! Launched in
        2019, it has delivered over 70 million+ products and even during the
        Covid-19 Lockdown. You can shop anything you want, including daily
        groceries and medicines. In short, a one-stop shop for everything that
        you need and craved for!
      </p>
      <br />
      <h1 className="catchy-tagline">What are you waiting for?!</h1>
      <Link to="/products">
        <button className="shopping-btn">Let's go Shopping!</button>
      </Link>
    </div>
  );
}

export default Home;
