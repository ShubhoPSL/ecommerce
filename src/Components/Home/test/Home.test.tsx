import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { RecoilRoot } from "recoil";
import Home from "../Home";

describe("Home Component is being tested", () => {
  const NewHome = () => {
    return (
      <RecoilRoot>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </RecoilRoot>
    );
  };
  test("Should have text about ShopKart", () => {
    render(<NewHome />);
    const paragraph = screen.getByText(
      /ShopKart is the fastest growing e-commerce platform of US! Launched in 2019, it has delivered over 70 million products also during the Covid-19 Lockdown. You can shop anything you want, including daily groceries and medicines. In short, a one-stop shop for everything that you need and craved for!/i
    );
    expect(paragraph).toBeInTheDocument();
  });
  test("Should render tagline", () => {
    render(<NewHome />);
    expect(
      screen.getByRole("heading", { name: "What are you waiting for?!" })
    ).toBeInTheDocument();
  });

  test("Should have shopping button", () => {
    render(<NewHome />);
    expect(
      screen.getByRole("button", { name: "Let's go Shopping!" })
    ).toBeInTheDocument();
  });

  test("should match with snapshot of Home module", () => {
    const tree = renderer.create(<NewHome />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
