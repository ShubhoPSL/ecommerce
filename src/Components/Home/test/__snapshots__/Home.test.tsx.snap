// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component is being tested should match with snapshot of Home module 1`] = `
<div
  className="home-container"
>
  <p
    className="home-body"
  >
    ShopKart is the fastest growing e-commerce platform of US! Launched in 2019, it has delivered over 70 million products also during the Covid-19 Lockdown. You can shop anything you want, including daily groceries and medicines. In short, a one-stop shop for everything that you need and craved for!
  </p>
  <br />
  <h1
    className="catchy-tagline"
  >
    What are you waiting for?!
  </h1>
  <a
    href="/products"
    onClick={[Function]}
  >
    <button
      className="shopping-btn"
    >
      Let's go Shopping!
    </button>
  </a>
</div>
`;
